pkg/cnservice/server.go

```diff
@@ -217,6 +213,21 @@ func (s *service) handleRequest(
 	req morpc.Message,
 	_ uint64,
 	cs morpc.ClientSession) error {
+	msg, ok := req.(*pipeline.Message)
+	if !ok {
+		logutil.Errorf("cn server should receive *pipeline.Message, but get %v", req)
+		panic("cn server receive a message with unexpected type")
+	}
+	switch msg.GetSid() {
+	case pipeline.WaitingNext:
```

Comment:
I don't think this is a suitable solution. 
If there are n queries ( q1, q2...qn) on cn0 at the same time, they all use the same cn client. 
There is no guarantee that the two msgs of q1 will arrive continuously. There may be msgs of q2 or qn interspersed in the middle. and msgs of q1 and q2 may both be of waitNext type, and hold the same ID (ID is stream sender id?).

Reply:
oh, each query will open a new stream sender. my fault.
it's not a problem.